src/cocotb/__init__.py

```diff
@@ -296,6 +296,19 @@ def _process_plusargs() -> None:
                 plusargs[option[1:]] = True
 
 
+class PackageNamespace(SimpleNamespace):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def __getattr__(self, name):
+        if SIM_NAME == "nvc":
+            # NVC does not preserve the case of identifiers so make
+            # lookups in this container case-insensitive
+            return self.__getattribute__(name.upper())
+        else:
+            super().__getattr__(name)
```

Comment:
I don't know if I like this answer. But I don't really think it's your problem to solve.

VHDL names are case insensitive (christ why...), but in mixed language simulations it's hard to know whether package names are case insensitive or if certain packages come from VHDL or Verilog and whether the case matters. And this seems to be a larger problem for handle names during iteration.

IMO object discovery via iteration with the VHPI should return correctly cased names and this shuldn't be necessary. VHDL does have the concept of object declarations so there is a source of truth for the caseness of every object. IDK why VHPI allows case insensitive operation, it was made at least 20 years after they realized what a bad idea case insensitive languages are.